service/topology_coordinator.cc

```diff
@@ -752,6 +753,87 @@ class topology_coordinator : public endpoint_lifecycle_subscriber {
         case global_topology_request::cleanup:
             co_await start_cleanup_on_dirty_nodes(std::move(guard), true);
             break;
+        case global_topology_request::keyspace_rf_change: {
+            while (true) {
+                auto tmptr = get_token_metadata_ptr();
+                sstring ks_name = *_topo_sm._topology.new_keyspace_rf_change_ks_name;
+                std::unordered_map<sstring, sstring> new_rf_per_dc = *_topo_sm._topology.new_keyspace_rf_change_rf_per_dc;
+                std::map<sstring, sstring> new_rf_per_dc_map{new_rf_per_dc.begin(), new_rf_per_dc.end()};
+                sstring req_uuid = *_topo_sm._topology.global_request_id;
+                std::vector<canonical_mutation> updates;
+                sstring error;
+                auto& ks = _db.find_keyspace(ks_name);
+                tablet_replica_calculation_result new_data{locator::tablet_map{8}, tablet_replica_calculation_status::success};
+
+                for (const auto& table : ks.metadata()->tables()) {
+                    auto repl_strategy = ks.get_replication_strategy_ptr()->maybe_as_tablet_aware();
+                    try {
+                        new_data = co_await service::calculate_tablet_replicas_for_new_rf(
+                                repl_strategy, table, tmptr, new_rf_per_dc_map);
+                    } catch(const std::exception& e) {
+                        new_data.status = tablet_replica_calculation_status::other;
+                        error = e.what();
+                        rtlogger.error("{}", error);
```

Comment:
this goes to show why it isn't a good idea to have both a status code in tablet_replica_calculation_result
and a possible error coming from an exceptional future.
too many error paths are a recipe for disaster.

Reply:
Also, the error message lacks context, I'm not sure what the error might say, but for example, if it's bad_alloc, how one understand what went wrong when it sees ` "std::bad_alloc" error message with no context in the log?

Reply:
Cc @KrzaQ

Reply:
I removed status codes and only left exceptions handling. 
The error logs have been merged, so that now I'm logging the `error` with additional context.